### PR TITLE
feat(@aws-amplify/auth): allow globalSignOut

### LINKED ISSUE #1525: Allow optional token invalidation when calling Auth.signOut()
> Do you want to request a feature or report a bug?

Request a feature

> What is the current behavior?

When `Auth.signOut()` is called, the session tokens _(access, refresh, id)_ are purged from local storage. However, the tokens themselves are still usabled.

> What is the expected behavior?

I would expect that when I end a session using `Auth.signOut()` that a request to invalidate the tokens is made, such that they are no longer valid for use. I understand that this is new behavior, thus I am suggesting that a new argument, `invalidateTokens` be added to `Auth.signOut()`, defaulted to false. This will preserve the existing behavior for current users, while allowing people to opt in to the new behavior. It might also be a good idea to expose this as a configuration option for `AuthClass`, so that I don't have to update all my calls to `Auth.signOut()`. This config option would be overridden by explicitly calling `Auth.signOut(true)`.

> Which versions of Amplify, and which browser / OS are affected by this issue? Did this work in previous versions?

1.0.6

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
fixes #1525 
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by undefobj** [comment]
Looks good. Let's also ensure that we add a bit more details around what this does to the documentation, specifically around AWS credentials not being impacted.

**Comment by powerful23** [comment]
@undefobj updated.

File: docs/media/authentication_guide.md [hunk 1]
```diff
@@ -169,6 +169,14 @@ import { Auth } from 'aws-amplify';
 Auth.signOut()
     .then(data => console.log(data))
     .catch(err => console.log(err));
+
+// By doing this, you are revoking all the auth tokens(id token, access token and refresh token)
+// which means the user is signed out from all the devices
+// Note: although the tokens are revoked, the AWS credentials will remain valid
```
Comment (mlabieniec) [inline docs/media/authentication_guide.md]: update this to:
Note: although the tokens are revoked, the AWS credentials will remain valid until they expire (which by default is 1 hour)

Comment (powerful23) [reply docs/media/authentication_guide.md]: done

Comment (madeeha96) [reply docs/media/authentication_guide.md]: how can we immediate logout all sessions? or reduce the time from 1 hour?

**Pull request merged**